lib/feature/browser/bottom_sheets/browser_bookmark_bottom_sheet.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:app/di/di.dart';
+import 'package:app/feature/browser/browser.dart';
+import 'package:app/generated/generated.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:share_plus/share_plus.dart';
+import 'package:ui_components_lib/ui_components_lib.dart';
+
+/// Helper function to show [BrowserBookmarkSheet].
+Future<void> showBrowserBookmarkSheet({
+  required BuildContext context,
+  required BrowserBookmarkItem item,
+}) {
+  return showCommonBottomSheet(
+    title: LocaleKeys.browserBookmarkSettings.tr(),
+    context: context,
+    body: (_, __) => BrowserBookmarkSheet(
+      item: item,
+    ),
+  );
+}
+
+class BrowserBookmarkSheet extends StatefulWidget {
+  const BrowserBookmarkSheet({
+    required this.item,
+    super.key,
+  });
+
+  final BrowserBookmarkItem item;
+
+  @override
+  State<BrowserBookmarkSheet> createState() => _BrowserBookmarkSheetState();
+}
+
+class _BrowserBookmarkSheetState extends State<BrowserBookmarkSheet> {
+  @override
+  Widget build(BuildContext context) {
+    final colors = context.themeStyle.colors;
+
+    return BlocProvider<BrowserBookmarksBloc>(
+      create: (context) => BrowserBookmarksBloc(
+        inject<BrowserBookmarksStorageService>(),
+      ),
+      child: Builder(
+        builder: (context) {
+          return ShapedContainerColumn(
+            margin: EdgeInsets.zero,
+            mainAxisSize: MainAxisSize.min,
+            separator: const CommonDivider(),
+            children: [
+              CommonListTile(
+                onPressed: () => _onSharePressed(context),
+                titleText: LocaleKeys.browserBookmarkShare.tr(),
+                trailing: CommonIconWidget.svg(
+                  svg: Assets.images.share.path,
+                  color: colors.textPrimary,
+                ),
+              ),
+              CommonListTile(
+                onPressed: () => _onRenamePressed(context),
+                titleText: LocaleKeys.browserBookmarkRename.tr(),
+                trailing: CommonIconWidget.svg(
+                  svg: Assets.images.edit.path,
+                  color: colors.textPrimary,
+                ),
+              ),
+              CommonListTile(
+                onPressed: () => _onDeletePressed(context),
+                titleText: LocaleKeys.browserBookmarkDelete.tr(),
+                trailing: CommonIconWidget.svg(
+                  svg: Assets.images.trash.path,
+                ),
+                contentColor: colors.alert,
+              ),
+            ],
+          );
+        },
+      ),
+    );
+  }
+
+  void _onSharePressed(BuildContext context) {
+    Navigator.of(context).pop();
+    final shareText = widget.item.url.toString();
+    Share.share(shareText);
+  }
+
+  void _onRenamePressed(BuildContext context) {
+    Navigator.of(context).pop();
+    showBrowserBookmarkRenameSheet(context: context, item: widget.item);
```

Comment:
If this works as expected?
It looks like here can be bug because you'r doing pop on first step and context after this action could be unachieved for `showBrowserBookmarkRenameSheet` method.

Reply:
🤟